public/interface.cpp

```diff
@@ -0,0 +1,272 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "interface.h"
+
+#if !defined ( _WIN32 )
```

Comment:
Can you use definitions from `build.h`?
And convert all #ifdef/#if defined to #if?